src/scorpio.h

```diff
@@ -244,13 +244,32 @@ piece and color
 /*
 distances
 */
-#define MAX(a, b)        (((a) > (b)) ? (a) : (b))
-#define MIN(a, b)        (((a) < (b)) ? (a) : (b))
+template<class T>
+const T& MAX_SCORPIO(const T& a, const T& b)
+{
+    return (a > b) ? b : a;

```

Comment:
I think the new issue we have is due to the `const T& a` references. We sometimes pass integer literals so that could break although I am not sure. It does seem to fix the crashes though.

Reply:
This turned out to be not straightforward after all. See for example this [discussion](https://stackoverflow.com/questions/12158582/how-to-implement-generic-max-function) for ways to do it the C++ way.  I do recall that in some cases I had to do an `#undef MAX` right before redefining it in the past, so maybe it will work if you just do that on your system ?

Reply:
Yes it is the first thing I tried, it didn't work. Then I renamed the defines, same error. The min has no problem, the issue is with the max. I already experienced this issue before, sometimes just including something like <vector> can cause this conflicting issue, even with defines that have different names! The only way I removed the error was to create the methods (which is a good thing in general because it is cleaner anyways ;) )

Reply:
Yes :) Do you have some accurate unit tests that you could launch?

Reply:
The tests for github actions are under tests/ and you can run that locally after building Scorpio.
I modified MAX and MIN to use the standard library versions as shown below and so far it doesn't crash with gcc or clang.
```C++
#define MAX_SCORPIO(a,b) std::max(a,b)
#define MIN_SCORPIO(a,b) std::min(a,b)
#define ABS(a)           std::abs(a)
#define f_distance(x,y)  ABS(file(x)-file(y))
#define r_distance(x,y)  ABS(rank(x)-rank(y))
#define distance(x,y)            MAX_SCORPIO(f_distance(x,y),r_distance(x,y))
#define DISTANCE(f1,r1,f2,r2)    MAX_SCORPIO(ABS((f1) - (f2)),ABS((r1) - (r2)))
```
Could we use the original names MAX and MIN in this version, because std::max is really not Scorpio's :)
I suppose MAX is already defined by the STL, but it would be good to know if the above modification along with the original names works on your system or not.
I am not able to reproduce the issue on my linux systems and github actions too so not sure what is causing it ..

Reply:
Hello Daniel :)
Unfortunately, this doesn't work, it was the first thing I tried. Basically on my system once you add the MAX define, even if you change its name, if the same line as the define contains parenthesis, it doesn't compile. :(

Reply:
This doesn't make sense to me. I can understand a name conflict i.e. MAX being defined somewhere else, but why would the compiler complain about some macro function that is not conflicting, and contains parenthesis.
Note that I primarily develop on ubuntu and never had this problem, and neither does github actions that is run on ubuntu -- so I need to be clear what is causing the problem you are seeing.